Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/D7AbsoluteCrossSections.py

```diff
@@ -202,97 +202,153 @@ def _cross_section_separation(self, ws, nMeasurements, nComponents):
-                    if nMeasurements == 6 and user_method == 'XYZ':
-                        # Total cross-section:
-                        dataY_total[spectrum] = (sigma_z_nsf + sigma_x_nsf + sigma_y_nsf + sigma_z_sf + sigma_x_sf + sigma_y_sf) / 3.0
-                        # Magnetic component
-                        dataY_nsf_magnetic[spectrum] = 2.0 * (2.0 * sigma_z_nsf - sigma_x_nsf - sigma_y_nsf)
-                        dataY_sf_magnetic[spectrum] = 2.0 * (-2.0 * sigma_z_sf + sigma_x_sf + sigma_y_sf)
-                        dataY_average_magnetic[spectrum] = 0.5 * (dataY_nsf_magnetic[spectrum] + dataY_sf_magnetic[spectrum])
-                        # Nuclear coherent component
-                        dataY_nuclear[spectrum] = (2.0*(sigma_x_nsf + sigma_y_nsf + sigma_z_nsf)
-                                                   - (sigma_x_sf + sigma_y_sf + sigma_z_sf)) / 6.0
-                        # Incoherent component
-                        dataY_incoherent[spectrum] = 0.5 * (sigma_x_sf + sigma_y_sf + sigma_z_sf) - dataY_average_magnetic[spectrum]
+            sigma_z_nsf = mtd[ws][entry_no]
+            sigma_z_sf = mtd[ws][entry_no+1]
+            total_cs = mtd[ws][entry_no].name() + '_Total'
+            nuclear_cs = mtd[ws][entry_no].name() + '_Coherent'
+            incoherent_cs = mtd[ws][entry_no].name() + '_Incoherent'
+            if nMeasurements == 2:
+                data_total = sigma_z_nsf + sigma_z_sf
+                RenameWorkspace(InputWorkspace=data_total, OutputWorkspace=total_cs)
+                separated_cs.append(total_cs)
+                data_nuclear = sigma_z_nsf - 0.5 * sigma_z_sf
+                RenameWorkspace(InputWorkspace=data_nuclear, OutputWorkspace=nuclear_cs)
+                separated_cs.append(nuclear_cs)
+                data_incoherent = 1.5 * sigma_z_sf
+                RenameWorkspace(InputWorkspace=data_incoherent, OutputWorkspace=incoherent_cs)
+                separated_cs.append(incoherent_cs)
+            elif nMeasurements == 6 or nMeasurements == 10:
+                sigma_y_nsf = mtd[ws][entry_no+2]
+                sigma_y_sf = mtd[ws][entry_no+3]
+                sigma_x_nsf = mtd[ws][entry_no+4]
+                sigma_x_sf = mtd[ws][entry_no+5]
+                average_magnetic_cs = mtd[ws][entry_no].name() + '_AverageMagnetic'
+                nsf_magnetic_cs = mtd[ws][entry_no].name() + '_NSFMagnetic'
+                sf_magnetic_cs = mtd[ws][entry_no].name() + '_SFMagnetic'
+                if nMeasurements == 6 and user_method == 'XYZ':
+                    # Total cross-section:
+                    data_total = (sigma_z_nsf + sigma_x_nsf + sigma_y_nsf + sigma_z_sf + sigma_x_sf + sigma_y_sf) / 3.0
+                    RenameWorkspace(InputWorkspace=data_total, OutputWorkspace=total_cs)
+                    separated_cs.append(total_cs)
+                    # Magnetic component
+                    dataY_nsf_magnetic = 2.0 * (2.0 * sigma_z_nsf - sigma_x_nsf - sigma_y_nsf)
```

Comment:
Thanks for changing this. Now that everything here are workspaces, could you rename dataY_... to just data_...?
The same appears in around line 320 as well.